public/mcdoc/create.mcdoc

```diff
@@ -33,168 +78,156 @@ type ItemOrTag = (
 	}
 )
 
+type ItemOrTagWithCount = (
+	struct {
+		item: string,
+		count?: int @ 1..,
+	} | struct {
+		tag: string,
+		count?: int @ 1..,
+	}
+)
+
 type FluidOrTag = (
 	struct {
 		fluid: string,
+		#[since="1.21.1"]
+		type: "fluid_stack",
 		amount: int @ 1..,
+		#[until="1.20.2"]
 		nbt?: NBT,
+		#[since="1.21.1"]
+		components?: DataComponentPatch,
 	} | struct {
+		#[until="1.20.2"]
 		fluidTag: string,
+		#[since="1.21.1"]
+		fluid_tag: string,
+		#[since="1.21.1"]
+		type: "fluid_tag",
 		amount: int @ 1..,
+		#[until="1.20.2"]
 		nbt?: NBT,
+		#[since="1.21.1"]
+		components?: DataComponentPatch,
 	}
 )
 
+dispatch create:recipes[create:compacting] to struct {
+	ingredients: [(ItemOrTagWithCount | FluidOrTag)] @ 1..,
+	/// Doesnt work properly
+	/// The options are id and id, it should item and fluid
+	results: [(ItemWithCount | Fluid)] @ 1,
```

Comment:
This is a pretty tricky union. What the generator does for unions of multiple structs, is always look at the first field to differenciate them. This works for vanilla cases like text components. You want to differenciate essentially between:
`{id:"apple",count:4}` and `{id:"water",amount:3}`. My immediate thought was that you could use `count` and `amount` instead as the first field. But `count` is optional, meaning that `{id:"apple"}` would be valid.

Really this indicates to me that Create's types are flawed itself. I don't have a good solution here if these types are correct.